tests/components/yale/test_config_flow.py

```diff
@@ -1,403 +1,205 @@
-
-
-async def test_form_reauth(hass: HomeAssistant) -> None:
-    """Test reauthenticate."""
-
-    entry = MockConfigEntry(
-        domain=DOMAIN,
-        data={
-            CONF_LOGIN_METHOD: "email",
-            CONF_USERNAME: "my@email.tld",
-            CONF_PASSWORD: "test-password",
-            CONF_INSTALL_ID: None,
-            CONF_TIMEOUT: 10,
-            CONF_ACCESS_TOKEN_CACHE_FILE: ".my@email.tld.august.conf",
+        "homeassistant.components.yale.async_setup_entry",
+        return_value=True,
+    ) as mock_setup:
+        await hass.config_entries.flow.async_configure(result["flow_id"])
+
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+    assert len(mock_setup.mock_calls) == 1
+    entry = hass.config_entries.async_entries(DOMAIN)[0]
+    assert entry.unique_id == USER_ID
+
+
+@pytest.mark.usefixtures("client_credentials")
+@pytest.mark.usefixtures("current_request_with_host")
+async def test_reauth(
+    hass: HomeAssistant,
+    hass_client_no_auth: ClientSessionGenerator,
+    aioclient_mock: AiohttpClientMocker,
+    mock_config_entry: MockConfigEntry,
+    reauth_jwt: str,
+) -> None:
+    """Test the reauthentication case updates the existing config entry."""
+
+    mock_config_entry.add_to_hass(hass)
+
+    mock_config_entry.async_start_reauth(hass)
+    await hass.async_block_till_done()
+
+    flows = hass.config_entries.flow.async_progress()
+    assert len(flows) == 1
+    result = flows[0]
+    assert result["step_id"] == "auth"
+
+    state = config_entry_oauth2_flow._encode_jwt(
+        hass,
+        {
+            "flow_id": result["flow_id"],
+            "redirect_uri": "https://example.com/auth/external/callback",
         },
-        unique_id="my@email.tld",
     )
-    entry.add_to_hass(hass)
-
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": config_entries.SOURCE_REAUTH}, data=entry.data
+    client = await hass_client_no_auth()
+    resp = await client.get(f"/auth/external/callback?code=abcd&state={state}")
+    assert resp.status == 200
+    assert resp.headers["content-type"] == "text/html; charset=utf-8"
+
+    aioclient_mock.post(
+        OAUTH2_TOKEN,
+        json={
+            "access_token": reauth_jwt,
+            "expires_in": 86399,
+            "refresh_token": "mock-refresh-token",
+            "user_id": USER_ID,
+            "token_type": "Bearer",
+            "expires_at": 1697753347,
+        },
     )
-    assert result["type"] is FlowResultType.FORM
-    assert result["errors"] == {}
-
-    with (
-        patch(
-            "homeassistant.components.yale.config_flow.YaleGateway.async_authenticate",
-            return_value=True,
-        ),
-        patch(
-            "homeassistant.components.yale.async_setup_entry",
-            return_value=True,
-        ) as mock_setup_entry,
+    with patch(
+        "homeassistant.components.yale.async_setup_entry",
+        return_value=True,
     ):
```

Comment:
Consider creating a common fixture for this

Reply:
061361c34a93c1ee9bf0f544e806bba3ba572009